File: B/PHP/Hardware.php
<?php

class Hardware extends Product{
    private $brand = 0;
    private $model = 0;

    public function __construct($brand = 0, $model = 0){
        $this->brand = $brand;
        $this->model = $model;
    }

    public function setBrand($brand){
        $this->brand = $brand;
    }

    public function getBrand(){
        return $this->brand;
    }

    public function setModel($model){
        $this->model = $model;
    }

    public function getModel(){
        return $this->model;
    }
}

?>

File: B/C++/Hardware.cpp
#include "Product.cpp"

class Hardware : public Product{
    private:
        string brand;
        string model;
    public:
        Hardware(){
            this->brand = "";
            this->model = "";
        }

        void setBrand(string b){
            this->brand = b;
        }

        string getBrand(){
            return this->brand;
        }

        void setModel(string m){
            this->model = m;
        }

        string getModel(){
            return this->model;
        }
};